controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -941,14 +986,48 @@ func (pp *portPublisher) unsubscribe(listener EndpointUpdateListener) {
 	pp.metrics.setSubscribers(len(pp.listeners))
 }
 
+func (pp *portPublisher) updateServer(server *v1beta1.Server, selector labels.Selector, isAdd bool) {
+	for _, address := range pp.addresses.Addresses {
+		if address.Pod != nil && selector.Matches(labels.Set(address.Pod.Labels)) {
+			var portMatch bool
+			switch server.Spec.Port.Type {
+			case intstr.Int:
+				if server.Spec.Port.IntVal == int32(address.Port) {
+					portMatch = true
+				}
+			case intstr.String:
+				for _, c := range address.Pod.Spec.Containers {
+					for _, p := range c.Ports {
+						if p.ContainerPort == int32(address.Port) && p.Name == server.Spec.Port.StrVal {
+							portMatch = true
+						}
+					}
+				}
+			default:
+				continue
+			}
+			if portMatch {
+				if isAdd && server.Spec.ProxyProtocol == opaqueProtocol {
+					address.OpaqueProtocol = true
+				} else {
+					address.OpaqueProtocol = false
+				}
```

Comment:
Because we change `OpaqueProtocol` field, the `Address`s in `AddressSet`s need to be pointers—this why that that change has been made.

We could instead keep it by value and do something like:

```go
for id, address := range pp.addresses.Addresses {
    ...
    address.OpaqueProtocol = true
    pp.addresses.Addresses[id] = address
```

But it felt like it was the right move to make them pointers anyway.